src/components/Navigation/Navigation.js

```diff
@@ -1,57 +1,90 @@
 import React from 'react';
 import './navigation.css';
 import { useLocation, Link } from 'react-router-dom';
-import {useSelector} from "react-redux";
+import { useDispatch, useSelector } from 'react-redux';
 import { useState, useEffect } from 'react';
 
 import { findUnreadNotificationsForUser } from '../../services/notifications-service';
+import { setNotifications } from '../../redux/userSlice';
+import { socket } from '../../services/socket-config';
 
 /**
  * Displays the main navigation menu of the app.
  */
 function Navigation() {
   const { pathname } = useLocation();
-
-  const [notifications, setNotifications] = useState([]);
+  const dispatch = useDispatch();
+  const notifications = useSelector((state) => state.user.notifications);
   const [setError] = useState();
 
   const authUser = useSelector((state) => state.user.data);
 
   // find all the unread notifications for a given user
   const findUnreadNotifications = async () => {
-      const res = await findUnreadNotificationsForUser(authUser.id);
-      if (res.error) {
-          return setError(
-              'We ran into an issue. Please try again later.'
-          );
-      }
-      setNotifications(res);
+    const res = await findUnreadNotificationsForUser(authUser.id);
+    if (res.error) {
+      return setError('We ran into an issue. Please try again later.');
+    }
+    dispatch(setNotifications(res));
   };
   useEffect(() => {
```

Comment:
Should this useEffect be removed, since there is also one after listenForNewNotificationsOnSocket()?